src/main/java/xbot/parser/Parser.java

```diff
@@ -0,0 +1,209 @@
+package xbot.parser;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+import xbot.TaskList;
+import xbot.XBotException;
+import xbot.storage.Storage;
+import xbot.task.Deadline;
+import xbot.task.Event;
+import xbot.task.Task;
+import xbot.task.ToDo;
+import xbot.ui.Ui;
+
+/**
+ * The Parser class provides utility methods to parse user input and tasks
+ * in the XBot application. It also handles the processing of user commands.
+ */
+public class Parser {
+
+    /**
+     * Parses a line of text representing a task from storage and returns the corresponding Task object.
+     *
+     * @param line The line of text to parse.
+     * @return The Task object represented by the text, or null if the format is unknown.
+     */
+    public static Task parseTask(String line) {
+        String[] parts = line.split(" \\| ");
+        if (parts.length >= 3) {
+            String type = parts[0].trim();
+            boolean isDone = parts[1].trim().equals("1");
+            String description = parts[2].trim();
+            switch (type) {
+            case "T":
+                Task todo = new ToDo(description);
+                if (isDone) {
+                    todo.setIsDone();
+                }
+                return todo;
+            case "D":
+                String deadline = parts[3].trim();
+                Task deadlineTask = new Deadline(description, deadline);
+                if (isDone) {
+                    deadlineTask.setIsDone();
+                }
+                return deadlineTask;
+            case "E":
+                String from = parts[3].trim();
+                String to = parts[4].trim();
+                Task eventTask = new Event(description, from, to);
+                if (isDone) {
+                    eventTask.setIsDone();
+                }
+                return eventTask;
+            default:
+                System.out.println("Unknown task type: " + type);
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Checks if a given date string is in a valid format.
+     *
+     * @param date The date string to validate.
+     * @return True if the date string matches one of the supported formats, false otherwise.
+     */
+    public static boolean isValidDateFormat(String date) {
+        List<String> formats = new ArrayList<>();
+        formats.add("yyyy-MM-dd");
+        formats.add("d/M/yyyy");
+        formats.add("d/M/yyyy HHmm");
+
+        for (String format : formats) {
```

Comment:
I like the way you saved the timing variable (for Deadline and Event type) as String. It creates the flexibily to especially when only the date provided for as a deadline or start/end of an event.